webui/react/src/pages/TrialDetails.tsx

```diff
@@ -24,11 +29,62 @@ const TrialDetailsComp: React.FC = () => {
   const trialId = parseInt(trialIdParam);
   const [ trial, triggerTrialRequest ] =
     useRestApi<TrialDetailsParams, TrialDetails>(getTrialDetails, { id: trialId });
+  const [ ContModalState, setContModalState ] = useState<CreateExpModalState>(
+    { config: 'Loading', visible: false },
+  );
+
```

Comment:
**suggestion:** separate these into two different `useState`. They are related in that they are both fed into the modal but that's about it. Changes in visibility and changes in config shouldn't alter each other.

Reply:
I see that, I can separate them. let's chat about it. Wouldn't react be smart enough to not cause rerenders if only one of these changes? On the other hand, I remember adding each individual hook adds an extra rerender.